runtime/drivers/athena/olap.go

```diff
@@ -345,74 +332,105 @@ func convertValue(colType, val string) (any, error) {
 		return f, nil
 	case "boolean":
 		return strconv.ParseBool(val)
-	case "date", "time", "time with time zone", "timestamp", "timestamp with time zone":
-		vv, err := scanTime(val)
-		if !vv.Valid {
-			return nil, fmt.Errorf("invalid time value")
+	case "date":
+		t, err := time.Parse(time.DateOnly, val)
+		if err != nil {
+			return nil, err
 		}
-		return vv.Time, err
+		return t, nil
+	case "time":
+		t, err := time.Parse(time.TimeOnly, val)
+		if err != nil {
+			return nil, err
+		}
+		return t, nil
+	case "time with time zone":
+		t, err := parseAthenaTimeWithLocation(time.TimeOnly, val)
+		if err != nil {
+			return nil, err
+		}
+		return t, nil
+	case "timestamp":
+		t, err := time.Parse(time.DateTime, val)
+		if err != nil {
+			return nil, err
+		}
+		return t, nil
+	case "timestamp with time zone":
+		t, err := parseAthenaTimeWithLocation(time.DateTime, val)
+		if err != nil {
+			return nil, err
+		}
+		return t, err
 	// for all other datatypes return string value directly
 	case "json", "char", "varchar", "varbinary", "row", "string", "binary",
 		"struct", "interval year to month", "interval day to second", "decimal",
 		"ipaddress", "array", "map", "unknown":
 		return val, nil
 	default:
-		return nil, fmt.Errorf("unknown type %q with value %q", colType, val)
+		return val, nil
 	}
 }
 
-// athenaTime represents a time.Time value that can be null.
-// The athenaTime supports Athena's Date, Time and Timestamp data types,
-// with or without time zone.
-type athenaTime struct {
-	Time  time.Time
-	Valid bool
-}
-
-var timeLayouts = []string{
-	"2006-01-02",
-	"15:04:05.000",
-	"2006-01-02 15:04:05.000000000",
-	"2006-01-02 15:04:05.000000",
-	"2006-01-02 15:04:05.000",
-}
-
-func scanTime(vv string) (athenaTime, error) {
-	parts := strings.Split(vv, " ")
-	if len(parts) > 1 && !unicode.IsDigit(rune(parts[len(parts)-1][0])) {
-		return parseAthenaTimeWithLocation(vv)
+func parseAthenaTimeWithLocation(layout, v string) (time.Time, error) {
+	idx := strings.LastIndexAny(v, "+-")
+	if idx == -1 {
```

Comment:
Did you check that Athena's formats can't be handled with Go's time formats? This library has a comparatively simpler implementation: https://github.com/segmentio/go-athena/blob/master/value.go

But it's not a well maintained library, so perhaps it's just the library that doesn't have good handling.